compute/backup/manage-backups.py

```diff
@@ -0,0 +1,152 @@
+    # add to list.
+
+    # Global reasons
+
+    if len(items) < minimum_number:
+        print "Fewer than %d snapshots, not deleting any" % minimum_number
+        return result
+
+    # Item-specific reasons
+
+    for item in _items[1:]: #always skip newest snapshot
+
+        item_timestamp = iso8601.parse_date(item['creationTimestamp'])
+
+        if now - item_timestamp < one_week:
+            print "Snapshot '%s' too new, not deleting." % item['name']
+            continue
+
+        if item_timestamp.weekday() == 1 and now - item_timestamp < three_months:
+            print "Snapshot '%s' is weekly timestamp and too new, not deleting." % item['name']
+            continue
+
+        if item_timestamp.day == 1 and now - item_timestamp < one_year:
+            print "Snapshot '%s' is monthly timestamp and too new, not deleting." % item['name']
+            continue
+
+        print "Adding snapshot '%s' to the delete list" % item['name']
+        result.append(item)
+
+    return result
+
+def create_snapshot(compute,project,disk,dry_run):
+
+    now = datetime.datetime.now(pytz.utc)
+    name = "%s-%s" % (disk,now.strftime('%Y-%m-%d'))
+    zone = zone_from_disk(disk)
+    print "Creating snapshot '%s' in zone '%s'" % (disk,zone)
+    if not dry_run:
+        result = compute.disks().createSnapshot(project=project, disk=disk, body={"name":name}, zone=zone).execute()
+
+def delete_snapshots(compute,project,snapshots,dry_run):
+    for snapshot in snapshots:
+        print "Deleting snapshot '%s'" % snapshot['name']
+        if not dry_run:
+            result = compute.snapshots().delete(project=project, snapshot=snapshot['name']).execute()
+
+def zone_from_disk(disk):
+    return DISK_ZONE_MAP[disk]
+
+def update_snapshots(compute,project,disk,dry_run):
+
+    filter = "name eq %s-[0-9]{4}-[0-9]{2}-[0-9]{2}" % disk
+    result = list_snapshots(compute,project,filter=filter)
+
+    if not result.has_key('items'):
+        print "Disk '%s' has no snapshots. Possibly it's new or you have a typo." % disk
+        snapshot_p = True
+        items_to_delete = []
+    else:
+        snapshot_p = should_snapshot(result['items'],datetime.timedelta(days=1))
+        items_to_delete = deletable_items(result['items'])
+
+    if snapshot_p:
+        create_snapshot(compute,project,disk,dry_run)
+
+    if len(items_to_delete):
+        delete_snapshots(compute,project,items_to_deelete,dry_run)
+
+def main(args):
+
+    disks = []
+    for diskzone in args.disk:
+        disk, zone = diskzone.split(',')
+        DISK_ZONE_MAP[disk] = zone
+        disks.append(disk)
+
+    credentials = GoogleCredentials.get_application_default()
+    compute = discovery.build('compute', 'v1', credentials=credentials)
+    project = args.project
+
+    for disk in disks:
+        update_snapshots(compute,project,disk,args.dry_run)
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description='Make and manage disk snapshots.')
```

Comment:
Pull the description from the module docstring as shown [here](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/storage/api/list_objects.py#L83).